<!DOCTYPE html>
<html>
<head>
  <title>2015-04-02 - The Low MQ Story</title>
  <meta charset="utf-8">
  <meta name="description" content="2015-04-02 - The Low MQ Story">
  <meta name="author" content="Cui Lab">
  <meta name="generator" content="slidify" />
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta http-equiv="X-UA-Compatible" content="chrome=1">
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/default.css" media="all" >
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/phone.css" 
    media="only screen and (max-device-width: 480px)" >
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/slidify.css" >
  <link rel="stylesheet" href="libraries/highlighters/highlight.js/css/tomorrow.css" />
  <base target="_blank"> <!-- This amazingness opens all links in a new tab. -->  
  
  <!-- Grab CDN jQuery, fall back to local if offline -->
  <script src="http://ajax.aspnetcdn.com/ajax/jQuery/jquery-1.7.min.js"></script>
  <script>window.jQuery || document.write('<script src="libraries/widgets/quiz/js/jquery.js"><\/script>')</script> 
  <script data-main="libraries/frameworks/io2012/js/slides" 
    src="libraries/frameworks/io2012/js/require-1.0.8.min.js">
  </script>
  
  

</head>
<body style="opacity: 0">
  <slides class="layout-widescreen">
    
    <!-- LOGO SLIDE -->
        <slide class="title-slide segue nobackground">
  <hgroup class="auto-fadein">
    <h1>2015-04-02 - The Low MQ Story</h1>
    <h2></h2>
    <p>Cui Lab<br/></p>
  </hgroup>
  <article></article>  
</slide>
    

    <!-- SLIDES -->
    <slide class="" id="slide-1" style="background:;">
  <hgroup>
    <h2>GiaB chr18-chr22</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>Isolated 1Mbp regions in the 10 samples</li>
<li>Removed regions with low (&lt; 4000000 reads) or high (&gt; 6500000 reads) coverage</li>
<li>Among the remaining 258/304 regions, read mapping quality sum to read counts ratio determined</li>
<li>Considered (10,15,20,25,50) lowest quality regions to corresponding highest quality regions</li>
<li>Compared how many times each SNP caller ranked highest in HM(Sens,PPV)</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-2" style="background:;">
  <hgroup>
    <h2>GiaB HM Low qual regions</h2>
  </hgroup>
  <article data-timings="">
    <p>Lowest 10</p>

<ul>
<li>MultiGeMS (6), SAMtools (2), VarScan (2)</li>
</ul>

<p>Lowest 15</p>

<ul>
<li>MultiGeMS (9), SAMtools (3), VarScan (3)</li>
</ul>

<p>Lowest 20</p>

<ul>
<li>MultiGeMS (10), SAMtools (6), VarScan (4)</li>
</ul>

<p>Lowest 25 </p>

<ul>
<li>MultiGeMS (14), SAMtools (7), VarScan (4)</li>
</ul>

<p>Lowest 50</p>

<ul>
<li>MultiGeMS (16), SAMtools (22), VarScan (12)</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-3" style="background:;">
  <hgroup>
    <h2>GiaB HM High qual regions</h2>
  </hgroup>
  <article data-timings="">
    <p>Highest 10</p>

<ul>
<li>SAMtools (10)</li>
</ul>

<p>Highest 15</p>

<ul>
<li>SAMtools (15)</li>
</ul>

<p>Highest 20</p>

<ul>
<li>SAMtools (19), VarScan (1)</li>
</ul>

<p>Highest 25 </p>

<ul>
<li>SAMtools (24), VarScan (1)</li>
</ul>

<p>Highest 50</p>

<ul>
<li>SAMtools (46), VarScan (4)</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-4" style="background:;">
  <hgroup>
    <h2>2015-03-18 Meeting with Prof. Cui</h2>
  </hgroup>
  <article data-timings="">
    <p>Recall the original objectives</p>

<ol>
<li>Check EM algorithm output on site MultiGeMS &quot;should&quot; have called</li>
<li>Analyze true SNP sites quantiles (seperated by quality scores)</li>
<li>Meta-analysis of 1Mbp quality regions of varying mapping quality</li>
<li>Start using Wei&#39;s implementation of MultiGeMS (to-do)</li>
</ol>

<p>From the results, add to these objectives</p>

<ol>
<li>Prof. Cui agrees that these results should be in the paper, but would also like the results from the UPennMed dataset (in which we might fix the positives and not use PPV).</li>
</ol>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-5" style="background:;">
  <hgroup>
    <h2>GiaB 20 MQ SC HM</h2>
  </hgroup>
  <article data-timings="">
    <p>82.13 90.46 92.46</p>

<table><thead>
<tr>
<th>Software</th>
<th>FreeBayes</th>
<th>GATK</th>
<th>SAMtools</th>
<th>MultiGeMS</th>
<th>VarScan</th>
</tr>
</thead><tbody>
<tr>
<td>Region 1</td>
<td>0.34821</td>
<td>0.18329</td>
<td>0.48582</td>
<td>0.44252</td>
<td>0.44371</td>
</tr>
<tr>
<td>Region 2</td>
<td>0.7564</td>
<td>0.49341</td>
<td>0.8509</td>
<td><font color="red">0.85136</font></td>
<td>0.83011</td>
</tr>
<tr>
<td>Region 3</td>
<td>0.97795</td>
<td>0.94855</td>
<td>0.98236</td>
<td>0.97623</td>
<td>0.97935</td>
</tr>
<tr>
<td>Region 4</td>
<td>0.99034</td>
<td>0.98657</td>
<td>0.99169</td>
<td>0.98867</td>
<td>0.98982</td>
</tr>
</tbody></table>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-6" style="background:;">
  <hgroup>
    <h2>GiaB 21 MQ SW TPR</h2>
  </hgroup>
  <article data-timings="">
    <p>71.78 90.32 91.66</p>

<table><thead>
<tr>
<th>Software</th>
<th>FreeBayes</th>
<th>GATK</th>
<th>SAMtools</th>
<th>MultiGeMS</th>
<th>VarScan</th>
</tr>
</thead><tbody>
<tr>
<td>Region 1</td>
<td>0.76019</td>
<td>0.66037</td>
<td>0.8293</td>
<td><font color="red">0.84347</font></td>
<td>0.79268</td>
</tr>
<tr>
<td>Region 2</td>
<td>0.90647</td>
<td>0.77961</td>
<td>0.95246</td>
<td>0.6378</td>
<td>0.93</td>
</tr>
<tr>
<td>Region 3</td>
<td>0.96554</td>
<td>0.93626</td>
<td>0.96845</td>
<td>0.92056</td>
<td>0.97832</td>
</tr>
<tr>
<td>Region 4</td>
<td>0.98406</td>
<td>0.96864</td>
<td>0.97347</td>
<td>0.96737</td>
<td>0.98205</td>
</tr>
</tbody></table>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-7" style="background:;">
  <hgroup>
    <h2>GiaB 22 MQ SW TPR</h2>
  </hgroup>
  <article data-timings="">
    <p>71.345 89.69 91.345</p>

<table><thead>
<tr>
<th>Software</th>
<th>FreeBayes</th>
<th>GATK</th>
<th>SAMtools</th>
<th>MultiGeMS</th>
<th>VarScan</th>
</tr>
</thead><tbody>
<tr>
<td>Region 1</td>
<td>0.76976</td>
<td>0.68707</td>
<td>0.79169</td>
<td><font color="red">0.8095</font></td>
<td>0.74418</td>
</tr>
<tr>
<td>Region 2</td>
<td>0.9096</td>
<td>0.89599</td>
<td>0.95408</td>
<td>0.85569</td>
<td>0.94627</td>
</tr>
<tr>
<td>Region 3</td>
<td>0.96895</td>
<td>0.96882</td>
<td>0.97899</td>
<td>0.95396</td>
<td>0.981</td>
</tr>
<tr>
<td>Region 4</td>
<td>0.98506</td>
<td>0.97493</td>
<td>0.97711</td>
<td>0.97239</td>
<td>0.98454</td>
</tr>
</tbody></table>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-8" style="background:;">
  <hgroup>
    <h2>GiaB 22 MQ SW HM</h2>
  </hgroup>
  <article data-timings="">
    <p>68.915 84.83 88.915</p>

<table><thead>
<tr>
<th>Software</th>
<th>FreeBayes</th>
<th>GATK</th>
<th>SAMtools</th>
<th>MultiGeMS</th>
<th>VarScan</th>
</tr>
</thead><tbody>
<tr>
<td>Region 1</td>
<td>0.42985</td>
<td>0.32536</td>
<td>0.46759</td>
<td><font color="red">0.50831</font></td>
<td>0.46694</td>
</tr>
<tr>
<td>Region 2</td>
<td>0.4304</td>
<td>0.22919</td>
<td>0.60155</td>
<td>0.5781</td>
<td>0.5821</td>
</tr>
<tr>
<td>Region 3</td>
<td>0.79038</td>
<td>0.53492</td>
<td>0.87326</td>
<td>0.84394</td>
<td>0.86132</td>
</tr>
<tr>
<td>Region 4</td>
<td>0.96578</td>
<td>0.93274</td>
<td>0.97112</td>
<td>0.96783</td>
<td>0.97005</td>
</tr>
</tbody></table>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-9" style="background:;">
  <hgroup>
    <h2>GiaB MQ SW HM</h2>
  </hgroup>
  <article data-timings="">
    <p>69.46 85.92 89.46</p>

<table><thead>
<tr>
<th>Software</th>
<th>FreeBayes</th>
<th>GATK</th>
<th>SAMtools</th>
<th>MultiGeMS</th>
<th>VarScan</th>
</tr>
</thead><tbody>
<tr>
<td>Region 1</td>
<td>0.4362</td>
<td>0.29069</td>
<td>0.48735</td>
<td><font color="red">0.49534</font></td>
<td>0.48233</td>
</tr>
<tr>
<td>Region 2</td>
<td>0.46265</td>
<td>0.2447</td>
<td>0.62348</td>
<td>0.5602</td>
<td>0.59205</td>
</tr>
<tr>
<td>Region 3</td>
<td>0.83107</td>
<td>0.60107</td>
<td>0.88526</td>
<td>0.86229</td>
<td>0.87573</td>
</tr>
<tr>
<td>Region 4</td>
<td>0.97628</td>
<td>0.94943</td>
<td>0.98029</td>
<td>0.97566</td>
<td>0.97824</td>
</tr>
</tbody></table>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-10" style="background:;">
  <hgroup>
    <h2>UPennMed chr19-chr22</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>Isolated 1Mbp regions in the 34 samples</li>
<li>Removed regions with low (&lt; 5500000 reads) or high (&gt; 12500000 reads) coverage</li>
<li>Among the remaining 187/225 regions, read mapping quality sum to read counts ratio determined</li>
<li>Considered (5,10,15,20,25,50) lowest quality regions to corresponding highest quality regions</li>
<li>Initially, compared how many times each SNP caller ranked highest in HM(Sens,PPV)</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-11" style="background:;">
  <hgroup>
    <h2>UPennMed HM</h2>
  </hgroup>
  <article data-timings="">
    <p>Upon closer examination, there is only nice advantage when looking at the 5 regions lowest in MQ</p>

<p>Lowest 5</p>

<ul>
<li>MultiGeMS (4), <em>No SNP Tie</em> (1)</li>
</ul>

<p>Highest 5</p>

<ul>
<li>FreeBayes (5)</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-12" style="background:;">
  <hgroup>
    <h2>Mu value comparison</h2>
  </hgroup>
  <article data-timings="">
    <p>In the regions of low mapping quality, where MultiGeMS ranks well, how do the mu values compare between sites that only MultiGeMS calls, compared to sites where MultiGeMS does not call, but other SNP callers do?</p>

<p>GiaB dataset, chrom 22, mq &lt; 71.345, 1762 sites: those called by all SNP callers (1685), in addition to those called exclusively by MultiGeMS (77)</p>

<p>Only result of interest: wilcox.test(snpmu0,notsnpmu0) p-value &lt; 2.2e-16</p>

<p>That is, the mu0 (first mu parameter) values from the set of &quot;true&quot; SNPs are significantly different from those of the &quot;false&quot; SNPs (see email Re: The dataset Wed, 25 Mar 2015 10:26:52).</p>

<p>Additionally, it was often seen that mu0 and mu1 differed significantly.</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-13" style="background:;">
  <hgroup>
    <h2>2015-03-25 Meeting with Prof. Cui</h2>
  </hgroup>
  <article data-timings="">
    <p><strong>Current priorities</strong></p>

<ol>
<li>Get started on running the entire GiaB dataset (with an extra 1TB of storage for 2-4 weeks) and use the whole-genome results in the paper</li>
<li>Design and run a simulation which will support the findings of the real data analysis (which is that MultiGeMS does well in low MQ data)</li>
<li>Since the unique aspect of MultiGeMS is the mu variable, select a site where the estimation of mu is advantageous: MultiGeMS makes a correct call when mu is estimated, yet an incorrect call, with results like all the other SNP callers, when not (setting mu = 0)</li>
</ol>

<p><strong>Done</strong></p>

<ul>
<li>All the tested SNP callers are fairly up-to-date, the oldest release being SAMtools 0.1.19 which is about 2 years old </li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-14" style="background:;">
  <hgroup>
    <h2>Test Wei&#39;s MultiGeMS code</h2>
  </hgroup>
  <article data-timings="">
    <p><strong>GiaB chr22 results, with step size 0.1: different from results from Xiaoquan&#39;s code</strong></p>

<pre><code class="R">length(xssnps) # 71514
length(wcsnps) # 67909, already nearly a 4,000 SNP count difference
length(setdiff(xssnps,wcsnps)) # 4039
length(setdiff(wcsnps,xssnps)) #  434
# sites in the intersection of the XS and WC SNP results
intersectsites &lt;- sort(unique(intersect(xssnps,wcsnps))) 
length(intersectsites) # 67475

# Percent of identical lFDR values at the sites of intersection sites, 
# when rounding to 4, 3, 2 and 1 values
length(which(round4df[,2] == round4df[,3]))/length(intersectsites) # 0.823979
length(which(round3df[,2] == round3df[,3]))/length(intersectsites) # 0.906973
length(which(round2df[,2] == round2df[,3]))/length(intersectsites) # 0.957303
length(which(round1df[,2] == round1df[,3]))/length(intersectsites) # 0.989611
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-15" style="background:;">
  <hgroup>
    <h2>GiaB: MQ and Coverage per Chr</h2>
  </hgroup>
  <article data-timings="">
    <p>Questionable QualiMap results</p>

<table><thead>
<tr>
<th>Chr</th>
<th>Cov Mean</th>
<th>Cov SD</th>
<th>MQ Mean</th>
</tr>
</thead><tbody>
<tr>
<td>18</td>
<td>12.64</td>
<td>114.22</td>
<td>1.51</td>
</tr>
<tr>
<td>19</td>
<td>8.96</td>
<td>187.10</td>
<td>1.22</td>
</tr>
<tr>
<td>20</td>
<td>9.97</td>
<td>73.92</td>
<td>1.24</td>
</tr>
<tr>
<td>21</td>
<td>6.38</td>
<td>91.09</td>
<td>0.80</td>
</tr>
<tr>
<td>22</td>
<td>5.50</td>
<td>53.20</td>
<td>0.80</td>
</tr>
</tbody></table>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-16" style="background:;">
  <hgroup>
    <h2>Test Wei&#39;s MultiGeMS code, part 2</h2>
  </hgroup>
  <article data-timings="">
    <p><strong>GiaB chr22 results, with step size 0.01: very similar to results from Xiaoquan&#39;s code</strong></p>

<table><thead>
<tr>
<th>Chr</th>
<th>XS</th>
<th>WC</th>
<th>Intersection</th>
</tr>
</thead><tbody>
<tr>
<td>18</td>
<td>146577</td>
<td>146561</td>
<td>146515</td>
</tr>
<tr>
<td>19</td>
<td>120849</td>
<td>120855</td>
<td>120838</td>
</tr>
<tr>
<td>20</td>
<td>112471</td>
<td>112470</td>
<td>112388</td>
</tr>
<tr>
<td>21</td>
<td>92042</td>
<td>92057</td>
<td>91713</td>
</tr>
<tr>
<td>22</td>
<td>71517</td>
<td>71514</td>
<td>71511</td>
</tr>
</tbody></table>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-17" style="background:;">
  <hgroup>
    <h2>Recall GiaB HM results</h2>
  </hgroup>
  <article data-timings="">
    <p>Chr18</p>

<table><thead>
<tr>
<th></th>
<th>SNP Count</th>
<th>Sens</th>
<th>PPV</th>
<th>F HM</th>
</tr>
</thead><tbody>
<tr>
<td>SAMtools</td>
<td>141846</td>
<td>0.9898</td>
<td>0.9181</td>
<td>0.9526</td>
</tr>
<tr>
<td>VarScan</td>
<td>140486</td>
<td>0.9791</td>
<td>0.9170</td>
<td>0.9470</td>
</tr>
<tr>
<td>MultiGeMS</td>
<td>146577</td>
<td>0.9897</td>
<td>0.8884</td>
<td>0.9363</td>
</tr>
<tr>
<td>FreeBayes</td>
<td>146203</td>
<td>0.9750</td>
<td>0.8774</td>
<td>0.9237</td>
</tr>
<tr>
<td>GATK</td>
<td>194206</td>
<td>0.9968</td>
<td>0.6753</td>
<td>0.8051</td>
</tr>
</tbody></table>

<p>Chr22</p>

<table><thead>
<tr>
<th></th>
<th>SNP Count</th>
<th>Sens</th>
<th>PPV</th>
<th>F HM</th>
</tr>
</thead><tbody>
<tr>
<td>SAMtools</td>
<td>70507</td>
<td>0.9793</td>
<td>0.8241</td>
<td>0.8950</td>
</tr>
<tr>
<td>VarScan</td>
<td>69568</td>
<td>0.9711</td>
<td>0.8282</td>
<td>0.8940</td>
</tr>
<tr>
<td>MultiGeMS</td>
<td>71517</td>
<td>0.9834</td>
<td>0.8158</td>
<td>0.8918</td>
</tr>
<tr>
<td>FreeBayes</td>
<td>75761</td>
<td>0.9633</td>
<td>0.7544</td>
<td>0.8461</td>
</tr>
<tr>
<td>GATK</td>
<td>111018</td>
<td>0.9920</td>
<td>0.5301</td>
<td>0.6910</td>
</tr>
</tbody></table>

  </article>
  <!-- Presenter Notes -->
</slide>

    <slide class="backdrop"></slide>
  </slides>
  <div class="pagination pagination-small" id='io2012-ptoc' style="display:none;">
    <ul>
      <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=1 title='GiaB chr18-chr22'>
         1
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=2 title='GiaB HM Low qual regions'>
         2
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=3 title='GiaB HM High qual regions'>
         3
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=4 title='2015-03-18 Meeting with Prof. Cui'>
         4
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=5 title='GiaB 20 MQ SC HM'>
         5
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=6 title='GiaB 21 MQ SW TPR'>
         6
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=7 title='GiaB 22 MQ SW TPR'>
         7
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=8 title='GiaB 22 MQ SW HM'>
         8
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=9 title='GiaB MQ SW HM'>
         9
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=10 title='UPennMed chr19-chr22'>
         10
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=11 title='UPennMed HM'>
         11
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=12 title='Mu value comparison'>
         12
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=13 title='2015-03-25 Meeting with Prof. Cui'>
         13
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=14 title='Test Wei&#39;s MultiGeMS code'>
         14
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=15 title='GiaB: MQ and Coverage per Chr'>
         15
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=16 title='Test Wei&#39;s MultiGeMS code, part 2'>
         16
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=17 title='Recall GiaB HM results'>
         17
      </a>
    </li>
  </ul>
  </div>  <!--[if IE]>
    <script 
      src="http://ajax.googleapis.com/ajax/libs/chrome-frame/1/CFInstall.min.js">  
    </script>
    <script>CFInstall.check({mode: 'overlay'});</script>
  <![endif]-->
</body>
  <!-- Load Javascripts for Widgets -->
  
  <!-- LOAD HIGHLIGHTER JS FILES -->
  <script src="libraries/highlighters/highlight.js/highlight.pack.js"></script>
  <script>hljs.initHighlightingOnLoad();</script>
  <!-- DONE LOADING HIGHLIGHTER JS FILES -->
   
  </html>